lib/jms/connection.rb

```diff
@@ -178,11 +177,20 @@ def initialize(params = {})
       connection_factory = nil
       factory = options.delete(:factory)
       if factory
+        # If factory check if oracle is needed.
+        if (factory.include? 'AQjmsFactory')
```

Comment:
Traditionally `:factory` is meant to refer to the actual Java factory class, which for oracle is `oracle.jms.AQjmsFactory`.
I agree it is confusing with Oracle because the Oracle JMS driver does not follow the same approach as other JMS providers and requires a "special" class to translate the standard behavior into what is required for Oracle AQ.

The class `OracleAQConnectionFactory` is intended to be completely hidden from the caller and is purely to achieve compatibility with the other JMS providers.

Since the Oracle AQ driver is a special case, maybe we should change it to:
~~~ruby
if (factory.include?('AQjmsFactory') || factory.include?('OracleAQConnectionFactory'))
~~~
However, this would not be consistent with the factory class name when it specified via JNDI, which needs to be `oracle.jms.AQjmsInitialContextFactory`

Ideally anyone using the Oracle AQ JMS provider would not have to know about `OracleAQConnectionFactory` and they can just supply the regular JMS class name of `oracle.jms.AQjmsFactory`.